### PR TITLE
feat(baremetal-server-components): add network tile component

### PR DESCRIPTION
**Pull request by radireddy**
ref: MANAGER-6934

Signed-off-by: Ravindra Adireddy <ravindra.adireddy@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/baremetal-server-components`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-6934
| License          | BSD 3-Clause

## Description

Added baremetal server network tile component

### CODE REVIEW

**Review by frenautvh — APPROVED** [review]

File: packages/manager/modules/bm-server-components/src/general-info-tile/template.html [hunk 1]
```diff
@@ -9,7 +9,6 @@
     >
         <oui-action-menu
             data-ng-if="::$ctrl.showNameActions()"
-            aria-label="{{:: 'menu_action_label' | translate:{ t0: ('server_configuration_displayname' | translate) } }}"
```
Comment (marie-j) [inline packages/manager/modules/bm-server-components/src/general-info-tile/template.html:12]: Already handled by the component ?

File: packages/manager/modules/bm-server-components/src/network/controller.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      .finally(() => {
+        this.loading = false;
+      });
+  }
+
+  loadVrackInfos() {
+    return this.getVrack(this.server.name)
+      .then((results) => {
+        const promises = results.map((vrack) =>
+          this.$http.get(`apiv6/vrack/${vrack}`).then(({ data }) => ({
```
Comment (marie-j) [inline packages/manager/modules/bm-server-components/src/network/controller.js:35]: Not possible to use iceberg with this ?

File: packages/manager/modules/bm-server-components/src/network/template.html [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                        data-ng-repeat="vrack in $ctrl.vrackInfos track by $index"
+                    >
+                        <a
+                            class="oui-link oui-link_icon"
+                            href="{{ ::$ctrl.getVrackUrl(vrack.serviceName) }}"
+                        >
+                            <span
+                                data-ng-bind="vrack.name || vrack.serviceName"
+                            ></span>
+                            (<span data-ng-bind="vrack.description"></span>)
```
Comment (marie-j) [inline packages/manager/modules/bm-server-components/src/network/template.html:54]: What if `vrakc.description` is undefined ?

File: packages/manager/modules/bm-server-components/src/network/controller.js (hunk 2 above)
Comment (radireddy) [reply packages/manager/modules/bm-server-components/src/network/controller.js:35]: iceberg does not return name and description required. It returns different data.

File: packages/manager/modules/bm-server-components/src/network/controller.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        return this.$q.all(promises);
+      })
+      .catch((error) => {
+        return this.handleError(error);
+      });
+  }
+
+  getVrack(serviceName) {
+    return this.$http
+      .get(`/dedicated/server/${serviceName}/vrack`)
```
Comment (ganeshkumar1989) [inline packages/manager/modules/bm-server-components/src/network/controller.js:49]: Why not use the iceberg capabilities to get all the vrack data in one API call?

**Pull request merged**